File: src/model/context.ts
import { CoreMessage } from 'ai';
import type { Message as UIMessage } from '../cli/types.js';

export function convertToLLMMessages(history: UIMessage[]): CoreMessage[] {
  return history
    .filter(msg => msg.type === 'user' || msg.type === 'assistant')
    .map(msg => ({ role: msg.type as 'user' | 'assistant', content: msg.text }));
}

File: inker-py/src/inker_py/context.py
"""Context management for conversation history."""

from .types import Message, MessageType, LLMMessage


def convert_to_llm_messages(history: list[Message]) -> list[LLMMessage]:
    """Convert UI messages to LLM messages."""
    result: list[LLMMessage] = []

    for msg in history:
        if msg.type == MessageType.USER:
            result.append(LLMMessage(role="user", content=msg.text))
        elif msg.type == MessageType.ASSISTANT:
            result.append(LLMMessage(role="assistant", content=msg.text))

    return result
